### ISSUE TITLE
Any rule for patch_shape?

### ISSUE THREAD

**Issue by Jun0Park** [ellisdg/3DUnetCNN#68]
1) The "patch_shape" for x, y, and z should be same number, right?
2) Your test example of "brats" has "image_shape" as (144, 144, 144).
Are those number related with "patch_shape" and "training_patch_start_offset"?
I mean (144-64)/16 is dividable. Doesn't it matter?

3) Do you think is it okay below configuration?

config["image_shape"] = (140, 140, 70)
config["patch_shape"] = (64, 64, 64)
config["training_patch_start_offset"] = (16, 16, 16)

or

config["image_shape"] = (256, 256, 70)
config["patch_shape"] = (64, 64, 64)
config["training_patch_start_offset"] = (10, 10, 10)

**Comment by ellisdg** [comment]
You should be able to set the image shape to whatever you like. The patch
shape however must be divisible by 2^(model depth). The default depth is 4,
so this means that the patch shape must be divisible by 16. So 32, 48, 64,
etc. should all work. I think you should be able to have a non cubic patch
shape as long as you turn off the permutation option. I haven't tried this,
but something like (64, 64, 32) should work as a patch shape.

On Dec 16, 2017 5:36 AM, "Jun0Park" <notifications@github.com> wrote:


   1.

   The "patch_shape" for x, y, and z should be same number, right?
   2.

   Your test example of "brats" has "image_shape" as (144, 144, 144).
   Are those number related with "patch_shape" and
   "training_patch_start_offset"?
   I mean (144-64)/16 is dividable. Doesn't it matter?
   3.

   Do you think is it okay below configuration?

config["image_shape"] = (140, 140, 70)
config["patch_shape"] = (64, 64, 64)
config["training_patch_start_offset"] = (16, 16, 16)

or

config["image_shape"] = (256, 256, 70)
config["patch_shape"] = (64, 64, 64)
config["training_patch_start_offset"] = (10, 10, 10)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/ellisdg/3DUnetCNN/issues/68>, or mute the thread

**Comment by Jun0Park** [comment]
Thank you for fast answer again!

My test result is below. training loss and dice coef looked okay, but validation loss and coef get decreased.
Do you have any expected problem for this?

0,0.026640455857,-0.0266404558242,0.0272111059251,-0.0272111059251

1,0.0259164407319,-0.0259164407427,0.0272214880044,-0.0272214884108

2,0.0269791242597,-0.0269791242597,0.029850484634,-0.0298504850404

3,0.0275371820061,-0.0275371822996,0.0284697250548,-0.0284697254612

4,0.0288512927774,-0.0288512927665,0.0299232098189,-0.0299232094125

5,0.031578258817,-0.0315782586871,0.0315628900447,-0.0315628900447

6,0.0349729545196,-0.0349729545414,0.0298091798682,-0.0298091798682

7,0.0471854929296,-0.0471854928422,0.020991277864,-0.020991277627

8,0.085965691713,-0.0859656917564,0.0171795272014,-0.0171795272014

9,0.126255393153,-0.126255392544,0.00349117742259,-0.00349117745222

**Comment by ellisdg** [comment]
The training loss looks good, but the validation scores aren't keeping up.
How big is your dataset?

**Comment by Jun0Park** [comment]
Data sets are 196 currently.
I have used validation_split as 0.9.
Do you think validation sets are too small?
